scipy/stats/stats.py

```diff
@@ -3313,50 +3309,61 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
 
     """
     a, axisout = _chk_asarray(a, axis)
+    if a.ndim > 2:
+        raise ValueError("spearmanr only handled 1-D or 2-D arrays")
 
-    a_contains_nan, nan_policy = _contains_nan(a, nan_policy)
-
-    if a_contains_nan:
-        a = ma.masked_invalid(a)
+    if b is None:
+        if a.ndim < 2:
+            raise ValueError("`spearmanr` needs at least 2 variables to compare")
```

Comment:
I guess adding a `Raises` section to the docstring might be appropriate eventually for the new exceptions here and above; that said, there's no mention of N-dimensional support (only 1 or 2 D) in the docstring anyway.

Reply:
In practice we never seem to use a `Raises` section; I think it's very rarely useful.